primitives/evm/src/account_provider.rs

```diff
@@ -22,14 +22,14 @@ pub trait AccountProvider {
 	/// for particular account.
 	type Index: AtLeast32Bit;
 
-	/// Creates a new account in accounts records.
+	/// Creates an EVM-managed account in accounts records, or marks the account as managed by EVM.
 	///
 	/// The account associated with new created address EVM.
-	fn create_account(who: &Self::AccountId);
-	/// Removes an account from accounts records.
+	fn create_evm_managed_account(who: &Self::AccountId);
+	/// Removes an EVM-managed account from accounts records, or unmarks the account as managed by EVM.
 	///
 	/// The account associated with removed address from EVM.
-	fn remove_account(who: &Self::AccountId);
+	fn remove_evm_managed_account(who: &Self::AccountId);
```

Comment:
Another iteration of investigations:
```suggestion
	fn on_account_delete(who: &Self::AccountId);
```

Reply:
Done